iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/BooleanLiteralImpl.java

```diff
@@ -21,11 +21,12 @@
 
 import java.util.List;
 import org.sonar.iac.arm.tree.api.BooleanLiteral;
+import org.sonar.iac.arm.tree.api.bicep.HasToken;
 import org.sonar.iac.arm.tree.api.bicep.SyntaxToken;
 import org.sonar.iac.arm.tree.impl.AbstractArmTreeImpl;
 import org.sonar.iac.common.api.tree.Tree;
 
-public class BooleanLiteralImpl extends AbstractArmTreeImpl implements BooleanLiteral {
+public class BooleanLiteralImpl extends AbstractArmTreeImpl implements BooleanLiteral, HasToken {
```

Comment:
The `BooleanLiteral` interface should extend the `HasToken` interface as it is strongly connected. No `BooleanLiteral` without a token.

Reply:
Should the json variant of `BooleanLiteralImpl` simply return null then? Or should we construct a new SyntaxToken around the boolean `value`?

Reply:
Result: Leave at is.